digitalocean/resource_digitalocean_monitor_alert.go

```diff
@@ -0,0 +1,291 @@
+package digitalocean
+
+import (
+	"context"
+	"log"
+
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceDigitalMonitorAlert() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDigitalOceanMonitorAlertCreate,
+		ReadContext:   resourceDigitalOceanMonitorAlertRead,
+		UpdateContext: resourceDigitalOceanMonitorAlertUpdate,
+		DeleteContext: resourceDigitalOceanMonitorAlertDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Required: true,
+			},
+
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					godo.DropletCPUUtilizationPercent,
+					godo.DropletMemoryUtilizationPercent,
+					godo.DropletDiskUtilizationPercent,
+					godo.DropletPublicOutboundBandwidthRate,
+					godo.DropletDiskReadRate,
+					godo.DropletDiskWriteRate,
+					godo.DropletOneMinuteLoadAverage,
+					godo.DropletFiveMinuteLoadAverage,
+					godo.DropletFifteenMinuteLoadAverage,
+					// these are available as constants ...
+					"v1/insights/droplet/public_inbound_bandwidth",
+					"v1/insights/droplet/private_outbound_bandwidth",
+					"v1/insights/droplet/private_inbound_bandwidth",
+				}, false),
+			},
+
+			"compare": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(godo.GreaterThan),
+					string(godo.LessThan),
+				}, false),
+				Description: "The comparison operator to use for value",
+			},
+
+			"description": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Description of the alert policy",
+			},
+
+			"enabled": {
+				Type:     schema.TypeBool,
+				Default:  true,
+				Required: true,
```

Comment:
Running the example (good example btw) and it looks like "Default cannot be set with Required", I'm thinking we drop the "default"

Reply:
Right, `Default` and `Required: true` can not be set at the same time:

https://www.terraform.io/docs/extend/schemas/schema-behaviors.html#default

`Required: true` enforces that the user always has it specified in their configuration file while providing a  `Default` implied that you do not need to explicitly set it in the configuration file as the default is used if not provided.

:+1:  I like using a default here as most users will want their alerts enabled by default.